annofabcli/annotation/change_annotation_properties.py

```diff
@@ -0,0 +1,286 @@
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class AnnotationDetailForCli(DataClassJsonMixin):
+    is_protected: Optional[bool] = None
+
+
+class ChangePropertiesOfAnnotation(AbstractCommandLineInterface):
+    """
+    アノテーションのプロパティを変更
+    """
+
+    COMMON_MESSAGE = "annofabcli annotation change_properties: error:"
+
+    def __init__(self, service: annofabapi.Resource, facade: AnnofabApiFacade, args: argparse.Namespace):
+        super().__init__(service, facade, args)
+        self.dump_annotation_obj = DumpAnnotation(service, facade, args)
+
+    def change_properties_for_task(
+        self,
+        project_id: str,
+        task_id: str,
+        annotation_query: AnnotationQuery,
+        properties: AnnotationDetailForCli,
+        force: bool = False,
+        backup_dir: Optional[Path] = None,
+    ) -> None:
+        """
+        タスクに対してアノテーションのプロパティを変更する。
+
+        Args:
+            project_id:
+            task_id:
+            annotation_query: 変更対象のアノテーションの検索条件
+            properties: 変更後のアノテーションのプロパティ
+            force: タスクのステータスによらず更新する
+            backup_dir: アノテーションをバックアップとして保存するディレクトリ。指定しない場合は、バックアップを取得しない。
+
+        """
+
+        def change_annotation_properties(
+            self, project_id: str, task_id: str, annotation_list: List[SingleAnnotation],
+            properties: AnnotationDetailForCli
+        ):
+            """
+            アノテーションのプロパティを変更する。
+
+            【注意】取扱注意
+
+            Args:
+                annotation_list: 変更対象のアノテーション一覧
+                properties: 変更後のプロパティ
+
+            Returns:
+                ``メソッドのレスポンス
+
+            """
+
+            def to_properties_from_cli(
+                annotation_list: List[SingleAnnotation], properties: AnnotationDetailForCli
+            ) -> List[Dict[str, Any]]:
+                annotations_for_api = []
+                annotation_details_by_input_data: Dict[str, List[Dict[str, Any]]] = {}
+
+                for annotation in annotation_list:
+                    input_data_id = annotation["input_data_id"]
+                    if input_data_id not in annotation_details_by_input_data:
+                        annotation_details_by_input_data[input_data_id] = []
+                    annotation_details_by_input_data[input_data_id].append(annotation["detail"])
+
+                for input_data_id, annotation_details in annotation_details_by_input_data.items():
+                    annotations, _ = self.service.api.get_editor_annotation(
+                        project_id, task_id, input_data_id
+                    )
+                    for idx, single_annotation in enumerate(annotations["details"]):
+                        for annotation_detail in annotation_details:
+                            if annotation_detail["annotation_id"] != single_annotation["annotation_id"]:
+                                continue
+                            annotations["details"][idx]["is_protected"] = properties.is_protected
```

Comment:
```suggestion
                            single_annotation["is_protected"] = properties.is_protected
```

こうすれば変数`idx`が不要になり、`enumuerate`を使わなくてよくなります。

Reply:
for文で渡される変数が値渡しだと思い込んでいました。参照渡しなのですね。覚えておきます。
ご指摘の通り修正しました。